### PR TITLE
Pci 1809 support lockfile only updates

### PR DESCRIPTION
**Pull request by aliculPix4D**
- refactor/simplify credentials handling
- enable setting `lockfile-only` option
- set the default `lockfile_only` values in case option is not set in `REPOSITORY_DATA` environmental variable

What is the `lockfile-only` option? Best to look at the PR examples: 
1) when the option is set to true, Pipfile is never updated, only Pipfile.lock can change
https://github.com/Pix4D/github-automation-playground/pull/114
This option is preferred by cloud team @jamesdynes 
2) when the option is set to false, dependencies in both files are changed, even the pinned ones if a new version exists
https://github.com/Pix4D/github-automation-playground/pull/115
The default behavior is set to `true` for Python (pip) module only, while in all other cases we keep the default Dependabot behavior (`false`). 
The PR in tomato will follow shortly.

In short: 
- pip module default `lockfile-only` option is set to true
- all other modules (i.e. docker) default `lockfile-only` option is set to false, keeping the same behavior as we have now 
- for each repository we can set this option `lockfile-only` in `terraform tomato` (github-automatio-pipeline). The set value will overwrite default value set here for that repo.

### CODE REVIEW

**Comment by Pix4D-Janus** [comment]
Your pipeline is set: https://builder.ci.pix4d.com/teams/developers/pipelines/dependabot-PCI-1809-support-lockfile-only-updates

File: pix4d-dependabot/spec/helpers/helper_dependabot_spec.rb [hunk 1]
```diff
@@ -364,3 +367,30 @@ def updated_dependency(name, updated_version, previous_version, file_name)
...
+      lockfile_only_defaults("any",
+                             { "lockfile_only" => "wrong" })
+    end .to raise_error(TypeError, "lockfile_only key should be boolean type")
+  end
+  context "for any module" do
+    let(:package_manager) { "docker" }
+    it "returns correct default value" do
+      expect(lockfile_only_defaults(package_manager, {})).to be false
+    end
+    it "it correctly replaces default value" do
```
Comment (odormond) [inline pix4d-dependabot/spec/helpers/helper_dependabot_spec.rb:383]:
```suggestion
    it "correctly replaces default value" do
```

File: pix4d-dependabot/spec/helpers/helper_dependabot_spec.rb [hunk 2]
```diff
@@ -364,3 +367,30 @@ def updated_dependency(name, updated_version, previous_version, file_name)
...
+      expect(lockfile_only_defaults(package_manager, { "module" => "docker", "lockfile_only" => true })).to be true
+    end
+  end
+  context "for pip module" do
+    let(:package_manager) { "pip" }
+    it "returns correct default value" do
+      expect(lockfile_only_defaults(package_manager, { "module" => "pip" })).to be true
+    end
+    it "it correctly replaces default value" do
```
Comment (odormond) [inline pix4d-dependabot/spec/helpers/helper_dependabot_spec.rb:392]:
```suggestion
    it "correctly replaces default value" do
```

**Review by odormond — APPROVED** [review]

**Review by Esysc — APPROVED** [review]
